src/Block/Service/AbstractBlockService.php

```diff
@@ -31,21 +32,25 @@
     protected $name;
 
     /**
-     * @var EngineInterface
+     * @var Environment
      */
     protected $templating;
 
     /**
-     * @param string          $name
-     * @param EngineInterface $templating
+     * @param string      $name
+     * @param Environment $templating
+     *
+     * @throws \TypeError when 2nd argument is not Twig\Environment
      */
-    public function __construct($name = null, EngineInterface $templating = null)
+    public function __construct($name, $templating)
     {
-        if (null === $name || null === $templating) {
+        if ($templating instanceof EngineInterface) {
             @trigger_error(
-                'The $name and $templating parameters will be required fields with the 4.0 release.',
+                'Using sonata.templating is deprecated since 4.0, use twig service directly.',
                 E_USER_DEPRECATED
             );
+        } elseif (!$templating instanceof Environment) {
+            throw new \TypeError('Argument 2 passed to '.get_class($this).'::__construct() must be instance of Twig\Environment, instance of '.get_class($templating).' given');
```

Comment:
Please linebreak this, use `sprintf`, `__METHOD__` and `Environment::class`

Reply:
Fixed